tests: fix Test_tagfiles: use Vim's 'tags' setting

blueyed opened the pull request:
Otherwise it might fail with an existing `tags` file upwards, e.g. in
neovim's source directory (which should be rather common).

blueyed committed:
tests: fix Test_tagfiles: use Vim's 'tags' setting

Otherwise it might fail with an existing `tags` file upwards, e.g. in
neovim's source directory (which should be rather common).

janlazo commented on src/nvim/testdir/test_taglist.vim:
```diff
@@ -76,6 +76,8 @@ func Test_tagsfile_without_trailing_newline()
 
 func Test_tagfiles()
+  " Nvim: defaults to "./tags;,tags", which might cause false positives.
+  set tags=./tags,tags
```
Can you move this in `src/nvim/testdir/setup.vim`?

blueyed replied:
Ok, done.

blueyed committed:
vim-patch:8.1.1546: in some tests 'tags' is set but not restored

Problem:    In some tests 'tags' is set but not restored. (Daniel Hahler)
Solution:   Restore 'tags'. (closes vim/vim#4535)
https://github.com/vim/vim/commit/0331faf71200a274377e1569fcda4f00c9cf3e2a

blueyed commented:
Included vim-patch:8.1.1546.

blueyed commented:
Travis failure is unrelated - will not restart it for now.
